cgi-bin/ljtestlib.pl

```diff
@@ -0,0 +1,83 @@
+package LJ;
+
+use strict;
+no warnings 'uninitialized';
+
+
+BEGIN {
+    use Test::MockObject;
+
+    my $mock = Test::MockObject->new();
+
+    sub fake_lang_ml {
+	my ( $code, $vars ) = @_;
+	my $aopts = $vars->{'aopts'};
+	if ( $code == "cleanhtml.error.markup.extra" ) {
+	    return "[<strong>Error:</strong> Irreparable invalid markup ('&lt;" . $aopts . "&gt;') in entry. Owner must fix manually. Raw contents below.]";
```

Comment:
nit: You don't need to break out of the string here, if you're using double-quotes, Perl will do variable interpolation for you so you can just drop the `$aopts` into the string.

```perl
return "[<strong>Error:</strong> Irreparable invalid markup ('&lt;$aopts&gt;') in entry. Owner must fix manually. Raw contents below.]";
```

Reply:
done